test/rekt/resources/pingsource/pingsource.go

```diff
@@ -57,7 +57,31 @@ func IsReady(name string, timings ...time.Duration) feature.StepFn {
 }
 
 // WithSink adds the sink related config to a PingSource spec.
-var WithSink = source.WithSink
+func WithSink(dest *duckv1.Destination) manifest.CfgFn {
+	return func(cfg map[string]interface{}) {
+		if _, set := cfg["sink"]; !set {
+			cfg["sink"] = map[string]interface{}{}
+		}
+		sink := cfg["sink"].(map[string]interface{})
+
+		uri := dest.URI
+		ref := dest.Ref
+
+		if uri != nil {
```

Comment:
Can they be both set, `uri` and `ref` ? Maybe worth adding a check that only one of them can be set? I think the Eventing components cannot set both. But totally sure.

Reply:
Or, we can probably leave it up to the caller to make sure only one is set. It's alright.

Reply:
Thanks for the details.